### DISCUSSION TITLE
Replacing assembly on circular view

### DISCUSSION THREAD

**Discussion by jrobinso** [GMOD/jbrowse-components#2474] [General]
Can I replace the chromosomes (sequence features) on a circular view dynamically?  The use case is user selects another genome assembly.   Here's what I tried, no errors but I can't seem to force it to repaint

```
        const regions = [ {"name": "1", "bpLength": 248956422}, {"name": "2", "bpLength": 242193529}]
        this.viewState.config.assembly.sequence.adapter.features.set(regions);
```

where the viewState was originally created with

```
                assembly: {
                    name: 'forIGV',
                    sequence: {
                        trackId: 'refSeqTrack',
                        type: 'ReferenceSequenceTrack',
                        adapter: {
                            type: 'FromConfigSequenceAdapter',
                            features: [ {"name": "1", "bpLength": 1}],
                        },
                    },
                }
```

If the question isn't clear I can provide a full example.   If this can't be done I suppose the alternative is to dispose of the current circular view and create a new one.    

I have another question about adding tracks but will create a new thread for that.

**Comment by jrobinso** [comment]
I think this is mostly answered here, looks to be the same question stated differently https://github.com/GMOD/jbrowse-components/discussions/2436.   However, after recreating the viewState do I need to destroy and recreate the React element (createElement), in other words build the DOM again, or is there a way to replace the viewState?

**Reply by cmdcolin** [reply]
you should be able to "just re-run the React.createElement call" AFAIK. React will notice that the reference to viewState is a different element, and re-run things accordingly

it might be good for us to make an example of this in our documentation

**Reply by jrobinso** [reply]
That works.   My wrapper code is at https://github.com/igvteam/circular-view.     This class will be the bridge between both igv.js and IGV desktop (via an electron wrapper). Its nearly complete with the exception of adding a track, which I have a workaround for.
